src/components/Button/Button.tsx

```diff
@@ -148,9 +148,8 @@ const ButtonComponentFn = <T extends ElementType = 'button'>(
               {processingSpinner || <Spinner size={size} />}
             </span>
           )}
-          {typeof children !== 'undefined' ? (
-            children
-          ) : (
+          {children ?? children}
```

Comment:
if `children` is simply rendering itself, then this is redundant.

Instead, simply render it like `{children}`, react will take care if its `null` or `undefined` and not render anything, else render the node.